From: Takashi Iwai <tiwai@suse.de>
Subject: kABI workaround for struct hdac_bus changes
Patch-mainline: Never, kABI workaround
References: git-fixes

A new field was added to struct hdac_bus by the patch
 patches.suse/ALSA-hda-add-member-to-store-ratio-for-stripe-contro.patch

The place has a few bytes of hole for padding and it can fit if the
field is changed to a byte.

Signed-off-by: Takashi Iwai <tiwai@suse.de>

---
 include/sound/hdaudio.h |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

--- a/include/sound/hdaudio.h
+++ b/include/sound/hdaudio.h
@@ -385,8 +385,11 @@ struct hdac_bus {
 	struct list_head hlink_list;
 	bool cmd_dma_state;
 
+	/* kABI workaround; adding a byte in the last padding hole */
+#ifndef __GENKSYMS__
 	/* factor used to derive STRIPE control value */
-	unsigned int sdo_limit;
+	unsigned char sdo_limit;
+#endif
 };
 
 int snd_hdac_bus_init(struct hdac_bus *bus, struct device *dev,
